smokes_mojo/lib/SmokesMojo/Controller/Api.pm

```diff
@@ -85,6 +88,58 @@ $msg
 EOS
 }
 
+sub post_coresmokedb_report ($self) {
+    my $req = $self->req;
+
+    my $schema = SmokeReports::Dbh->schema;
+    my $sdb = $schema->resultset("Perl5Smoke");
+    my $id = $sdb->get_column("report_id")->max;
+    $id = $id + 1;
+
+    my $json = Cpanel::JSON::XS->new->utf8;
+    my $report_data = $json->decode($req->body);
+    $report_data->{$_} = $report_data->{'sysinfo'}{$_} for keys %{ $report_data->{'sysinfo'} };
+    $report_data->{$_} = $report_data->{'config'}{$_} for keys %{ $report_data->{'config'} };
```

Comment:
There's no need for the loops, better as:
```
my $sysinfo = delete $report_data->{'sysinfo'};
$report_data->@{keys %$sysinfo} = values %$sysinfo;
my $config = delete $report_data->{'config'};
$report_data->@{keys %$config} = values %$config;
```

Reply:
Though they should be conditional on whether the keys are set - your sample report doesn't have a `config` key.